### PR TITLE
Issue #4227 Fix recipients when replying, from === acctEmail, not the first message in a thread, its subject starts with Re:

### LINKED ISSUE #4227: Live gmail test 'secure reply btn, reply draft' is failing
```
  mail.google.com - secure reply btn, reply draft
    ℹ (attempt 1 of 3) Retrying: AssertionError: expected 'ci.tests.gmail@flowcrypt.devlimon.monte@gmail.com' to equal 'limon.monte@gmail.com'
    ℹ (attempt 2 of 3) Retrying: AssertionError: expected 'ci.tests.gmail@flowcrypt.devlimon.monte@gmail.com' to equal 'limon.monte@gmail.com'
    ℹ (attempt 3 of 3) Failed:   AssertionError: expected 'ci.tests.gmail@flowcrypt.devlimon.monte@gmail.com' to equal 'limon.monte@gmail.com'
```

The latest green build was on Dec 24

### PR DESCRIPTION
**Pull request by limonte**
This PR aligns the behavior of Reply button with Gmail and when:

- replying to not the first message in the thread
- from header === acctEmail (from myself)

Then acctEmail shoudn't be present in reply to headers

close #4227 

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: test/source/tests/compose.ts [hunk 1]
```diff
@@ -1497,6 +1497,13 @@ export const defineComposeTests = (testVariant: TestVariant, testWithBrowser: Te
       await expectRecipientElements(composePage, { to: ['flowcrypt.compatibility@gmail.com', 'vladimir@flowcrypt.com'], cc: [], bcc: [] });
     }));
 
+    ava.default('compose - reply - from === acctEmail, not the first message in a thread, its subject starts with Re:', testWithBrowser('compatibility', async (t, browser) => {
+      const appendUrl = 'threadId=17d02296bccd4c5d&skipClickPrompt=___cu_false___&ignoreDraft=___cu_false___&replyMsgId=17d02296bccd4c5d';
```
Comment (rrrooommmaaa) [inline test/source/tests/compose.ts:1501]: I see threadId = '17d02296bccd4c5d', and messageId = '17d02296bccd4c5d'. Does this mean that this message is actually the first in the thread?

**Comment by rrrooommmaaa** [comment]
Also, isn't `GmailParser` used in gmail replacement only (that is, Gmail Live tests)?

File: test/source/tests/compose.ts (hunk 1 above)
Comment (limonte) [reply test/source/tests/compose.ts:1501]: Correct, it's the first one in the mock, but with `Re: ` in its subject, so it acts like the response in the thread.

**Comment by limonte** [comment]
> Also, isn't `GmailParser` used in gmail replacement only (that is, Gmail Live tests)?

Sorry, I didn't get this question, can you elaborate a bit?

**Comment by rrrooommmaaa** [comment]
> Sorry, I didn't get this question, can you elaborate a bit?

As found when working on another PR https://github.com/FlowCrypt/flowcrypt-browser/pull/4202#discussion_r774936765_, there are 2 parsers involved.
1) GmailParser is used on a gmail page when gmail-element-replacer is in action.
2) Mime parsing (`extension/js/common/core/mime.ts`) is used on our own settings/inbox page
So, I thought the first one is backed by Gmail Live tests (`gmail.ts`) and the second one is backed by all others tests(including `compose.ts` ).
Or am I wrong?

File: test/source/tests/compose.ts (hunk 1 above)
Comment (rrrooommmaaa) [reply test/source/tests/compose.ts:1501]: Ok, can we name the test as `compose - reply - subject starts with Re:` as this is the only thing that is tested here from my understanding?

**Comment by limonte** [comment]
> > Sorry, I didn't get this question, can you elaborate a bit?
> 
> As found when working on another PR [#4202 (comment)](https://github.com/FlowCrypt/flowcrypt-browser/pull/4202#discussion_r774936765)_, there are 2 parsers involved.
> 
> 1. GmailParser is used on a gmail page when gmail-element-replacer is in action.
> 2. Mime parsing (`extension/js/common/core/mime.ts`) is used on our own settings/inbox page
>    So, I thought the first one is backed by Gmail Live tests (`gmail.ts`) and the second one is backed by all others tests(including `compose.ts` ).
>    Or am I wrong?

GmailParser and this logic of determining `to` headers is used both in Gmail and our own Inbox page.

File: test/source/tests/compose.ts (hunk 1 above)
Comment (limonte) [reply test/source/tests/compose.ts:1501]: Done 👍

**Review by rrrooommmaaa — APPROVED** [review]

**Comment by rrrooommmaaa** [comment]
> GmailParser and this logic of determining `to` headers is used both in Gmail and our own Inbox page.

Thanks for clarification

**Pull request merged**